### DISCUSSION TITLE
show TopMost and split the volume bar into left and right channel

### DISCUSSION THREAD

**Discussion by mrchanshares** [File-New-Project/EarTrumpet#1105] [Ideas]
I want an option to 
1. allow the application to stay top Most.
2. instead of only show the volume level as one bar, allow it to show as left and right channel.

**Comment by mrchanshares** [comment]
Also, 
3. to make the _mixerWindow transparent.
Anyone can help?

**Comment by mrchanshares** [comment]
4. I want to make the _mixerWindow, which is of the WindowHolder class,  not close by hitting the Escape key accidently. How can I do that? Thanks

**Comment by mrchanshares** [comment]
It is very frustrated to have the project built one day and another day it fails to build. 
I probably will list the changes I made after cloning the project and have it running....

**Comment by mrchanshares** [comment]
Severity Code Description Project File Line Suppression State 
		Error CS8107 Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater. EarTrumpet C:\Users\a\source\repos\File-New-Project\EarTrumpet\EarTrumpet\Extensions\IPropertyStore Extensions.vs 20 Active
		
		reference:
		https://developercommunity.visualstudio.com/t/feature-default-literal-is-not-available-in-c-70-f/471188

Project > Build > Advanced... > Language version from C# latest major version (default) to either C# latest minor version (latest) or C# 7.1, C# 7.2, C# 7.3 fixes the error.

**Comment by mrchanshares** [comment]
After having visual studio 2017 installed, check the visual studio installer to see if there is further updates/patches.

tools -> get tools & features -> close the dialog box to see the visual studio installer

**Reply by riverar** [reply]
You should not be using Visual Studio 2017. (I see we recommended that in our docs, oops. I'm going to bump that up to at least Visual Studio 2019.)

**Reply by mrchanshares** [reply]
2019? Why not just go straight to 2022?
So you think visual studio 2022 is better? But I still face a lot of problems with it as well. 

Try starting everything from zero. Build it etc.

**Reply by riverar** [reply]
Yeah, good point. The `rafael/dotnet` branch already requires it (big .NET 6 migration). If it doesn't work well out of the box (as documented), please do file an issue. That's something we take seriously!

**Reply by mrchanshares** [reply]
That is a good start.
And I can get the earTrumpet project, the main part of the solution, compiled with visual studio 2022 running on Window 11 this week. So things are running smooth now.
